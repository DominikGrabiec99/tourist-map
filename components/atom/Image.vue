<template>
  <img class="image" data-test="image" :src="src" :alt="altText" />
</template>

<script lang="ts" setup>
withDefaults(
  defineProps<{
    src: string;
    altText?: string;
  }>(),
  {
    src: '',
    altText: 'ZdjÄ™cie',
  }
);
</script>
