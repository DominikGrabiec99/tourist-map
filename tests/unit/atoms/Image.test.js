import {
  describe,
  expect,
  it
} from 'vitest';
import {
  mount
} from '@vue/test-utils';
import Image from '../../../components/atom/Image.vue';

describe('Image.vue', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(Image, {
      props: {
        altText: 'Photo',
        src: '../../../public/favicon.png',
      },
    });
  });

  it('render image', () => {
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
    expect(wrapper.attributes('src')).toContain('../../../public/favicon.png');
    expect(wrapper.attributes('alt')).toContain('Photo');
  });
});