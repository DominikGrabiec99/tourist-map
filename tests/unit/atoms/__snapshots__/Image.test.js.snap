// Vitest Snapshot v1

exports[`Image.vue > render image 1`] = `
<img
  alt="Photo"
  class="image"
  data-test="image"
  src="../../../public/favicon.png"
/>
`;
